tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
+      aikido::statespace::InterpolatorPtr& _interpolator,
+      aikido::distance::DistanceMetricPtr& _dmetric,
+      aikido::constraint::SampleablePtr& _sampler,
+      aikido::constraint::TestablePtr& _collConstraint,
+      aikido::constraint::TestablePtr& _boundsConstraint,
+      aikido::constraint::ProjectablePtr& _boundsProjection,
+      Eigen::Vector3d _startPose, Eigen::Vector3d _goalPose          
+    )
+  {
+
+    auto startState = _stateSpace->createState();
+    auto subState1 = _stateSpace->getSubStateHandle<R3>(startState, 0);
+    subState1.setValue(_startPose);
+
+    auto goalState = _stateSpace->createState();
+    auto subState2 = _stateSpace->getSubStateHandle<R3>(goalState, 0);
+    subState2.setValue(_goalPose);
+
+    // Plan
+    auto traj = aikido::planner::ompl::planOMPL<ompl::geometric::RRTConnect>(
+        startState, goalState, _stateSpace, _interpolator, std::move(_dmetric),
+        std::move(_sampler), std::move(_collConstraint),
+        std::move(_boundsConstraint), std::move(_boundsProjection), 5.0, 0.1);
+
+    return traj;
+
+  }
+
+} // nampespace
+
+
+// Test that the start and goal positions do not change
+TEST_F(SimplifierTest, EndPointsRemainUnchanged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Original Trajectory
+  auto traj = planTrajForTest(stateSpace, interpolator, dmetric, sampler, collConstraint, 
+      boundsConstraint, boundsProjection, startPose, goalPose);
+
+  // Setup for simplification
+  aikido::distance::DistanceMetricPtr dmetric = aikido::distance::createDistanceMetric(stateSpace);
+  aikido::constraint::SampleablePtr sampler = aikido::constraint::createSampleableBounds(stateSpace, make_rng()); 
+  aikido::constraint::ProjectablePtr boundsProjection = aikido::constraint::createProjectableBounds(stateSpace);
+  aikido::constraint::TestablePtr boundsConstraint = aikido::constraint::createTestableBounds(stateSpace);
+  aikido::constraint::TestablePtr collConstraint = std::make_shared<MockTranslationalRobotConstraint>(
+        stateSpace, Eigen::Vector3d(-0.1, -0.1, -0.1),
+        Eigen::Vector3d(0.1, 0.1, 0.1));
+
+  auto simplifiedPair = aikido::planner::ompl::simplifyOMPL(
+      stateSpace, interpolator, std::move(dmetric), std::move(sampler), 
+      std::move(collConstraint), std::move(boundsConstraint), std::move(boundsProjection),
+      0.1, 5.0, 10, traj);
+
+  // Simplification results
+  auto simplifiedTraj = std::move(simplifiedPair.first);
+
+  // Check the first waypoint
+  auto s0 = stateSpace->createState();
+  simplifiedTraj->evaluate(0, s0);
+  auto r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(startPose));
+
+  // Check the last waypoint
+  simplifiedTraj->evaluate(simplifiedTraj->getDuration(), s0);
+  r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(goalPose));
```

Comment:
Replace this with the `EXPECT_EIGEN_EQUAL` macro defined in `eigen_tests.hpp`. It has the same effect, but will print a beautiful error message if there is a mismatch. 😍

The same comment applies to the other `EXPECT_TRUE(... isApprox ... )` calls in these tests.